src/fetch.js

```diff
@@ -44,10 +51,44 @@ function runFetch ({
   })
     .then(checkStatus)
     .then(createResponse)
-    .then((response) => [decrementFetches(), next(response)])
-    .catch((error) =>
-      createErrorResponse(error)
-        .then((response) => [decrementFetches(), fetchError(response), next(response, true)]))
+    .then(async (response) =>
+      (retry && await retry(response))
```

Comment:
`retry` is a function that receives that response and can return either a Boolean or a Promise that evaluates to a Boolean.